packages/odyssey-react/src/components/Modal/index.tsx

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { createContext, useContext, useMemo, useRef } from "react";
+import type { FunctionComponent, ReactElement, ReactNode, ReactText } from "react";
+import { createPortal } from "react-dom";
+import Button from "../Button";
+import type { ButtonVariants } from "../Button";
+import { useOid, useCx } from "../../utils";
+import { useKeypress, useOutsideClick } from '../../hooks';
+import useFocusTrap from '../../hooks/useFocusTrap';
+
+export type PropsModal = {
+  /**
+   * The modal content, should use the Static components provided by Modal (Modal.Header, Modal.Body and Modal.Footer)
+   */
+  children: ReactElement | ReactElement[],
+
+  /**
+   * The modal id attribute. Automatically generated if not provided.
+   */
+  id?: string,
+
+  /**
+   * Determines whether or not the modal is visible.
+   */
+  open: boolean,
+
+  /**
+   * Callback when the modal is opened.
+   */
+   onOpen?: () => void,
+
+  /**
+   * Callback when the modal is closed.
+   */
+  onClose: () => void,
+}
+  
+export type PropsModalHeader = { 
+  children: ReactText
+}
+
+export type PropsModalBody = { 
+  children: ReactNode
+}
+
+export type PropsModalFooter = { 
+  children: ReactNode
+}
+
+export type PropsModalButton = { 
+  close?: boolean,
+  children: ReactText,
+  variant?: ButtonVariants,
+  onClick?: () => void
+}
+
+export type StaticComponents = {
+  Header: FunctionComponent<PropsModalHeader>,
+  Body: FunctionComponent<PropsModalBody>,
+  Footer: FunctionComponent<PropsModalFooter>,
+  Button: FunctionComponent<PropsModalButton>
+}
+
+export const ModalContext = createContext<{ onClose: () => void}>({ onClose: () => void 0 });
```

Comment:
lets plan to move context and children components out of this file in a subsequent PR